### PR TITLE
Add details on git tags and rake publish command

### PR DESCRIPTION
**Pull request by kylelonghurst**

### CODE REVIEW

File: PUBLISHING.md [hunk 1]
```diff
@@ -4,11 +4,19 @@
 
 This document outlines the process for publishing the Smartsheet Live Data Connector User Guide onto Github Pages. The actual Smartsheet Live Data Connector documentation can be viewed [here](http://smartsheet-platform.github.io/odbc-docs). 
 
+## Why git version tags?
+Git tags are a nice way to keep track of the versions of the documentation. The version of the documentation should match the version of the Smartsheet Live Data Connector that it supports.
```
Comment (emilykoh) [inline PUBLISHING.md]: maybe add an example of how to git tag and explanation on how to use semantic versioning

ex. `git tag v1.0.1 after a patch`

Comment (kfansler) [reply PUBLISHING.md]: Agree. Add this syntax example.

File: PUBLISHING.md [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
+
+## Understanding the publish command
+The `rake publish` command bundles all of the resources in the `source/` directory to create static assets in a local directory called `bundle/`. These static assesst are then pushed into the `gh-pages` branch. Github pages hosts the contents of the `gh-pages` for public viewing. Only Admins and Owners of the repository have push access into the `gh-pages` repo, so `rake publish` command will fail for anyone else.
```
Comment (kfansler) [inline PUBLISHING.md]: Same typo as other doc: assesst should possibly be assets

**Review by kfansler — CHANGES_REQUESTED** [review]
Same two changes as the other.

**Review by kfansler — APPROVED** [review]

**Pull request merged**